rasa/importers/importer.py

```diff
@@ -271,3 +273,105 @@ async def get_nlu_data(self, language: Optional[Text] = "en") -> TrainingData:
         return reduce(
             lambda merged, other: merged.merge(other), nlu_data, TrainingData()
         )
+
+
+class E2EImporter(TrainingDataImporter):
+    """Importer which
+    - enhances the NLU training data with actions / user messages from the stories.
+    - adds potential end-to-end bot messages from stories as actions to the domain
+    """
+
+    def __init__(self, importer: TrainingDataImporter) -> None:
+        self._importer = importer
+        self._cached_stories: Optional[StoryGraph] = None
+
+    async def get_domain(self) -> Domain:
+        original, e2e_domain = await asyncio.gather(
+            self._importer.get_domain(), self._get_domain_with_e2e_actions()
+        )
+        return original.merge(e2e_domain)
+
+    async def _get_domain_with_e2e_actions(self) -> Domain:
+        from rasa.core.events import ActionExecuted
+
+        stories = await self.get_stories()
+
+        additional_e2e_action_names = []
+        for story_step in stories.story_steps:
+            additional_e2e_action_names += [
+                event.action_name
+                for event in story_step.events
+                if isinstance(event, ActionExecuted) and event.e2e_text
+            ]
+
+        return Domain(
+            [], [], [], {}, action_names=additional_e2e_action_names, form_names=[]
+        )
+
+    async def get_stories(
+        self,
+        interpreter: "NaturalLanguageInterpreter" = RegexInterpreter(),
+        template_variables: Optional[Dict] = None,
+        use_e2e: bool = False,
+        exclusion_percentage: Optional[int] = None,
+    ) -> StoryGraph:
+        if not self._cached_stories:
+            # Simple cache to avoid loading all of this multiple times
+            self._cached_stories = await self._importer.get_stories(
+                interpreter, template_variables, use_e2e, exclusion_percentage
+            )
+        return self._cached_stories
+
+    async def get_config(self) -> Dict:
+        return await self._importer.get_config()
+
+    async def get_nlu_data(self, language: Optional[Text] = "en") -> TrainingData:
```

Comment:
it can actually be `None` according to https://github.com/RasaHQ/rasa/blob/f41326d21763d41ff92b8c469e434ab90d071a4d/rasa/nlu/config.py#L173 

However, this helped me finding a bug in the code.

Reply:
👍